cameo/util.py

```diff
@@ -602,6 +602,32 @@ def reduce_reaction_set(reaction_set, groups):
     return result
 
 
+def decompose_reaction_groups(reaction_groups, reactions):
+    """
+
+    reaction_groups : list
+        A list with dictionaries (element: relative_coefficient)
+    reactions : list, set, tuple
+        A collection of reactions.
+
+    Returns
+    -------
+    list
+        A list of all possible group substitutions.
+
+    """
+    to_keep = []
+    to_replace = {}
+    for element in reactions:
+        for g in reaction_groups:
+            if element in g:
+                to_replace[element] = g.keys()
```

Comment:
There should probably be a break here. The if below could then be changed to an 'else' on the for loop